File: languages/rust/experiments/java_rust/ownership.java
class Ownership {
    public static void main(String[] args) {
        var item = new Item(10);

        var item_reference = item;

        Ownership.doSomething(item);
        Ownership.doSomething(item_reference);
    }

    public static void doSomething(Item item) {
        item.count = item.count + 1;
        System.out.println("Doing something with " + item.count + " items.");
    }
}

class Item {
    int count;

    Item(int startWith) {
        this.count = startWith;
    }
}

File: languages/rust/experiments/java_rust/ownership.rs
fn main() {
    let mut item = Item { count: 11 };
    let item_reference = &mut item;

    do_something(item_reference);
    do_something(&mut item);
}

fun do_something(item: &mut Item) {
    println!("Doing something with {} items.", item.count);
}

struct Item {
    count: i32,
}
